src/Compilers/CSharp/Test/Semantic/Semantics/NullableReferenceTypesTests.cs

```diff
@@ -115258,13 +115223,70 @@ class C : I<object>,
 {
 }
 
+interface I<T> {}
+";
+            var comp = CreateCompilation(source);
+            comp.VerifyDiagnostics();
+        }
+
+        [Fact, WorkItem(38168, "https://github.com/dotnet/roslyn/issues/38168")]
+        public void MethodTypeInference_ImplementedInterfaces_Direct_AnnotatedVersusUnannotated()
+        {
+            var source = @"
+#nullable enable warnings
+static class P
+{
+    static void M(C c)
+    {
+        c.Extension();
+    }
+
+    public static void Extension<T>(this I<T> source) => throw null!;
+}
+
+class C : I<object>, I<object?>
+{
+}
+
 interface I<T> {}
 ";
             var comp = CreateCompilation(source);
             comp.VerifyDiagnostics(
-                // (13,7): warning CS8645: 'I<object>' is already listed in the interface list on type 'C' with different nullability of reference types.
-                // class C : I<object>,
-                Diagnostic(ErrorCode.WRN_DuplicateInterfaceWithNullabilityMismatchInBaseList, "C").WithArguments("I<object>", "C").WithLocation(13, 7)
+                // (13,30): warning CS8632: The annotation for nullable reference types should only be used in code within a '#nullable' annotations context.
+                // class C : I<object>, I<object?>
```

Comment:
>I<object>, I<object?> [](start = 29, length = 21)

Should we report a warning even for an oblivious mismatch when it happens for interfaces directly present in the base list?

Reply:
I'm going for an approach mimicking tuple name differences. We could have special rules for nullability, but I'm not sure it helps or is worthwhile.
The email thread landed on warnings for all cases, which is also consistent with the general approach we took for nullability.
Do you think it should be an error?

---
In reply to: [325296374](https://github.com/dotnet/roslyn/pull/38563#discussion_r325296374) [](ancestors = 325296374)